From 2d1da21169a3a49b703940940c3fad991dbff11d Mon Sep 17 00:00:00 2001
From: krebbel <krebbel@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 12 Apr 2013 05:39:49 +0000
Subject: [PATCH] 2013-04-12  Andreas Krebbel  <Andreas.Krebbel@de.ibm.com>

	* ifcvt.c (end_ifcvt_sequence): Mark a and b for unsharing as
	well.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_8-branch@197840 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog |    5 +++++
 gcc/ifcvt.c   |    2 ++
 2 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index 719bc5b..6fbc1bb 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,8 @@
+2013-04-12  Andreas Krebbel  <Andreas.Krebbel@de.ibm.com>
+
+	* ifcvt.c (end_ifcvt_sequence): Mark a and b for unsharing as
+	well.
+
 2013-04-11  Marek Polacek  <polacek@redhat.com>
 
 	PR tree-optimization/48184
diff --git a/gcc/ifcvt.c b/gcc/ifcvt.c
index faea882..88967a6 100644
--- a/gcc/ifcvt.c
+++ b/gcc/ifcvt.c
@@ -964,6 +964,8 @@ end_ifcvt_sequence (struct noce_if_info *if_info)
 
   set_used_flags (if_info->x);
   set_used_flags (if_info->cond);
+  set_used_flags (if_info->a);
+  set_used_flags (if_info->b);
   unshare_all_rtl_in_chain (seq);
   end_sequence ();
 
-- 
1.7.1

